src/Algebra/Graph/AdjacencyMap/Extra.hs

```diff
@@ -43,3 +46,29 @@ shrinkSingle vert gr = AM.overlay (AM.removeVertex vert gr) inducedEdges
         | pre <- Set.toList (AM.preSet vert gr)
         , post <- Set.toList (AM.postSet vert gr)
         ]
+
+-- | Split graph into distinct sibling graphs via nodes that have no incoming edges.
+-- Returns 'Nothing' if graph is cyclic.
+--
+-- @
+-- splitGraph 'edges' [(1, 2), (2, 3), (4, 5)] = ['edges' [(1, 2), (2, 3)], 'edge' 4 5]
+-- @
+splitGraph :: Ord a => AM.AdjacencyMap a -> Maybe [AM.AdjacencyMap a]
```

Comment:
My biggest problem was actually determining the correct solution for a split that included cycles.  Since I couldn't, and the package.json graph cannot contain cycles, I did this instead.